hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncRequestFutureImpl.java

```diff
@@ -764,9 +763,14 @@ private void failAll(MultiAction actions, ServerName server, int numAttempt,
    * @param t          the throwable (if any) that caused the resubmit
    */
   private void receiveGlobalFailure(MultiAction rsActions, ServerName server, int numAttempt,
-    Throwable t, boolean clearServerCache) {
+    Throwable t) {
     errorsByServer.reportServerError(server);
     Retry canRetry = errorsByServer.canTryMore(numAttempt) ? Retry.YES : Retry.NO_RETRIES_EXHAUSTED;
+    boolean clearServerCache = false;
+
+    if (!(t instanceof RejectedExecutionException)) {
```

Comment:
Enforces the constraints added in https://issues.apache.org/jira/browse/HBASE-27491

Reply:
I think it would be better if you instead push `RejectedExecutionException` down into `ClientExceptionsUtil.isMetaClearingException`.

How about adding another collection of execution-exceptions for the family of various ExecutorService interaction errors, like is done with networking/connection exceptions?

Reply:
Noted, adding that